### PR TITLE
[WIP] Build `compare` feature

### PR DESCRIPTION
**Pull request by bushshrub**
Addresses #20

### CODE REVIEW

File: approx/core/api.py [hunk 1]
```diff
@@ -55,8 +56,13 @@ def compare(
         Useful statistical information
     """
     if metrics is None or len(metrics) == 0:
-        metrics = [Metric.LOSS]
-    runner = _CompareRunner([model, quantized_model], eval_loop, metrics)
+        metrics = [Metric.LOSS, Metric.ACCURACY]
```
Comment (bushshrub) [inline approx/core/api.py]: @sudomaze should we require accuracy here? Since this can't be measured on regression tasks

File: approx/core/api.py [hunk 2]
```diff
@@ -55,8 +56,13 @@ def compare(
...
+    for metric in metrics:
```
Comment (bushshrub) [inline approx/core/api.py]: We should probably add check here to see if len(set(metrics)) == len(metrics)

File: approx/core/api.py (hunk 1 above)
Comment (ma7dev) [reply approx/core/api.py]: This is just a placeholder at the moment, but I have an idea of using a `dict` which is the output of the `eval_loop` to make it dynamic. Here is an example:

```python
def eval_loop(...):
    ...
    return {
        "loss": loss_history,
        "accuracy": accuracy_history,
        ...
        "metric": METRIC_HISTORY
    }
```

File: approx/core/api.py (hunk 2 above)
Comment (ma7dev) [reply approx/core/api.py]: https://github.com/approx-ml/approx/pull/21#discussion_r934595742

File: approx/core/api.py (hunk 1 above)
Comment (bushshrub) [reply approx/core/api.py]: Yeah, using a List wasn't the brightest idea on my part

Comment (bushshrub) [reply approx/core/api.py]: Updated this in the latest commit

File: approx/core/api.py [hunk 3]
```diff
@@ -24,6 +26,53 @@ def auto_quantize(model: Any, pretrained: bool = True) -> Any:
...
+
+    Returns:
+        Useful statistical information
+    """
+    runner = _CompareRunner([model, quantized_model], test_loader, eval_loop)
+    return runner.run()
+
+
+def make_eval_loop(
+    model: Any, loss_fn: Callable, acc_fn: Optional[Callable[..., List[bool]]]
```
Comment (bushshrub) [inline approx/core/api.py]: Mypy failure

File: approx/core/backend/_backend.py [hunk 4]
```diff
@@ -76,6 +87,22 @@ def auto_quantize(self, model: Any, pretrained: bool) -> Any:
             "Currently not implemented for non-pretrained models"
         )
 
+    def generate_eval_loop(
+        self,
+        model: Any,
+        loss_fn: Callable,
```
Comment (bushshrub) [inline approx/core/backend/_backend.py]: Make proper `Protocol` subclass for this

**Pull request merged**